<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ase::MainWindow</class>
 <widget class="QMainWindow" name="ase::MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QStatusBar#statusBar
{
	border-top: 1px solid #B9B9B9;
}</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <property name="verticalSpacing">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>24</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">QTabBar::close-button
{
	image: url(:/images/tab_close.png);
}

QTabBar::close-button:hover
{
	image: url(:/images/tab_close_hover.png);
}

QTabBar::close-button:pressed
{
	image: url(:/images/tab_close_press.png);
}

QTabWidget:pane
{
}</string>
      </property>
      <property name="documentMode">
       <bool>false</bool>
      </property>
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Page</string>
       </attribute>
      </widget>
     </widget>
    </item>
    <item row="0" column="1" rowspan="2">
     <widget class="QTreeWidget" name="treeWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>181</width>
        <height>401</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">QTreeWidget
{
	border: 1px solid #909090;
	border-top: none;
	border-right: none;
	border-bottom: none;
}</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <attribute name="headerVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="headerStretchLastSection">
       <bool>true</bool>
      </attribute>
      <column>
       <property name="text">
        <string notr="true">1</string>
       </property>
      </column>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="handleWidth">
       <number>0</number>
      </property>
      <property name="childrenCollapsible">
       <bool>false</bool>
      </property>
      <widget class="ase::ASEEditor" name="plainTextEdit">
       <property name="minimumSize">
        <size>
         <width>610</width>
         <height>150</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">QPlainTextEdit#plainTextEdit
{
	border-top: 1px solid #B9B9B9;
	border-right: 1px solid #909090;
	border-left: none;
	border-bottom: none;
}

QPlainTextEdit#plainTextEdit
{
	background-color: #FFFFFF;
}</string>
       </property>
       <property name="lineWrapMode">
        <enum>QPlainTextEdit::NoWrap</enum>
       </property>
       <property name="tabStopWidth">
        <number>40</number>
       </property>
      </widget>
      <widget class="QFrame" name="frame">
       <property name="minimumSize">
        <size>
         <width>610</width>
         <height>100</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">QFrame#frame
{
	border-top: 1px solid #b9b9b9;
	border-right: 1px solid #909090;
	background-color: white;
}</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <property name="lineWidth">
        <number>0</number>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QToolBar" name="errorToolBar">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>34</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">QToolBar#errorToolBar
{
    background-color: #e9e9e9;
	border-bottom: 1px solid #b9b9b9;
}</string>
          </property>
          <addaction name="actionClearTable"/>
          <addaction name="actionSaveTable"/>
          <addaction name="actionCloseDebugger"/>
         </widget>
        </item>
        <item>
         <widget class="QTableWidget" name="tableWidget">
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Plain</enum>
          </property>
          <property name="lineWidth">
           <number>0</number>
          </property>
          <property name="sortingEnabled">
           <bool>true</bool>
          </property>
          <property name="rowCount">
           <number>0</number>
          </property>
          <property name="columnCount">
           <number>3</number>
          </property>
          <attribute name="horizontalHeaderVisible">
           <bool>true</bool>
          </attribute>
          <attribute name="horizontalHeaderCascadingSectionResizes">
           <bool>false</bool>
          </attribute>
          <attribute name="horizontalHeaderDefaultSectionSize">
           <number>50</number>
          </attribute>
          <attribute name="horizontalHeaderHighlightSections">
           <bool>true</bool>
          </attribute>
          <attribute name="horizontalHeaderMinimumSectionSize">
           <number>20</number>
          </attribute>
          <attribute name="horizontalHeaderShowSortIndicator" stdset="0">
           <bool>true</bool>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <attribute name="verticalHeaderVisible">
           <bool>false</bool>
          </attribute>
          <attribute name="verticalHeaderDefaultSectionSize">
           <number>20</number>
          </attribute>
          <attribute name="verticalHeaderMinimumSectionSize">
           <number>20</number>
          </attribute>
          <attribute name="verticalHeaderShowSortIndicator" stdset="0">
           <bool>false</bool>
          </attribute>
          <attribute name="verticalHeaderStretchLastSection">
           <bool>false</bool>
          </attribute>
          <column>
           <property name="text">
            <string/>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Line</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Message</string>
           </property>
           <property name="font">
            <font>
             <strikeout>false</strikeout>
            </font>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionSave_Script"/>
    <addaction name="actionSave_ROM"/>
    <addaction name="actionSave_As"/>
    <addaction name="separator"/>
    <addaction name="actionNew_Script"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="separator"/>
    <addaction name="actionCut"/>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
    <addaction name="actionDelete"/>
    <addaction name="separator"/>
    <addaction name="actionSelect_All"/>
    <addaction name="actionFind_Replace"/>
   </widget>
   <widget class="QMenu" name="menuProject">
    <property name="title">
     <string>Project</string>
    </property>
    <addaction name="actionOpen_Project"/>
    <addaction name="actionSave_Project"/>
    <addaction name="separator"/>
    <addaction name="actionNew_Project"/>
    <addaction name="separator"/>
    <addaction name="actionProperties"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionCompile_Script"/>
    <addaction name="actionDebug_Script"/>
    <addaction name="separator"/>
    <addaction name="actionDecompile_Script"/>
    <addaction name="actionDecompile_Levelscript"/>
    <addaction name="separator"/>
    <addaction name="actionCreate_text"/>
    <addaction name="actionCreate_movement"/>
   </widget>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>?</string>
    </property>
    <addaction name="actionHelp"/>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
    <addaction name="actionReal_time_script_validator"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuProject"/>
   <addaction name="menuTools"/>
   <addaction name="menuSettings"/>
   <addaction name="menu"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="iconSize">
    <size>
     <width>24</width>
     <height>24</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen_2"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionFind"/>
   <addaction name="actionCompile"/>
   <addaction name="separator"/>
   <addaction name="actionDecompile"/>
   <addaction name="actionLevelscript"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="../ASE.qrc">
     <normaloff>:/images/toolbar_open.png</normaloff>:/images/toolbar_open.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action">
   <property name="text">
    <string>-</string>
   </property>
  </action>
  <action name="actionSave_Script">
   <property name="text">
    <string>Save Script</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_ROM">
   <property name="icon">
    <iconset resource="../ASE.qrc">
     <normaloff>:/images/toolbar_save.png</normaloff>:/images/toolbar_save.png</iconset>
   </property>
   <property name="text">
    <string>Save ROM</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save As...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionNew_Script">
   <property name="text">
    <string>New Script...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="text">
    <string>Undo</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="text">
    <string>Redo</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="text">
    <string>Cut</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="text">
    <string>Copy</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="text">
    <string>Paste</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="text">
    <string>Delete</string>
   </property>
  </action>
  <action name="actionSelect_All">
   <property name="text">
    <string>Select All</string>
   </property>
  </action>
  <action name="actionFind_Replace">
   <property name="text">
    <string>Find/Replace</string>
   </property>
  </action>
  <action name="actionOpen_Project">
   <property name="text">
    <string>Open Project</string>
   </property>
  </action>
  <action name="actionSave_Project">
   <property name="text">
    <string>Save Project</string>
   </property>
  </action>
  <action name="actionNew_Project">
   <property name="text">
    <string>New Project</string>
   </property>
  </action>
  <action name="actionProperties">
   <property name="text">
    <string>Properties...</string>
   </property>
  </action>
  <action name="actionCompile_Script">
   <property name="text">
    <string>Compile Script</string>
   </property>
  </action>
  <action name="actionDebug_Script">
   <property name="text">
    <string>Debug Script</string>
   </property>
  </action>
  <action name="actionDecompile_Script">
   <property name="text">
    <string>Decompile Script</string>
   </property>
  </action>
  <action name="actionDecompile_Levelscript">
   <property name="text">
    <string>Decompile Levelscript</string>
   </property>
  </action>
  <action name="actionCreate_text">
   <property name="text">
    <string>Create text</string>
   </property>
  </action>
  <action name="actionCreate_movement">
   <property name="text">
    <string>Create movement</string>
   </property>
  </action>
  <action name="actionHelp">
   <property name="text">
    <string>Help</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="../ASE.qrc">
     <normaloff>:/images/info.png</normaloff>:/images/info.png</iconset>
   </property>
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionOpen_2">
   <property name="icon">
    <iconset resource="../ASE.qrc">
     <normaloff>:/images/toolbar_open.png</normaloff>:/images/toolbar_open.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="../ASE.qrc">
     <normaloff>:/images/toolbar_save.png</normaloff>:/images/toolbar_save.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
  </action>
  <action name="actionFind">
   <property name="icon">
    <iconset resource="../ASE.qrc">
     <normaloff>:/images/toolbar_find.png</normaloff>:/images/toolbar_find.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
  </action>
  <action name="actionCompile">
   <property name="icon">
    <iconset resource="../ASE.qrc">
     <normaloff>:/images/toolbar_comp.png</normaloff>:/images/toolbar_comp.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
  </action>
  <action name="actionDecompile">
   <property name="icon">
    <iconset resource="../ASE.qrc">
     <normaloff>:/images/toolbar_decomp.png</normaloff>:/images/toolbar_decomp.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
  </action>
  <action name="actionLevelscript">
   <property name="icon">
    <iconset resource="../ASE.qrc">
     <normaloff>:/images/toolbar_level.png</normaloff>:/images/toolbar_level.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
  </action>
  <action name="actionReal_time_script_validator">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Real-time script validator</string>
   </property>
  </action>
  <action name="actionClearTable">
   <property name="icon">
    <iconset resource="../ASE.qrc">
     <normaloff>:/images/table_delete.png</normaloff>:/images/table_delete.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
  </action>
  <action name="actionSaveTable">
   <property name="icon">
    <iconset resource="../ASE.qrc">
     <normaloff>:/images/table_save.png</normaloff>:/images/table_save.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
  </action>
  <action name="actionCloseDebugger">
   <property name="icon">
    <iconset resource="../ASE.qrc">
     <normaloff>:/images/close_misc.png</normaloff>:/images/close_misc.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ase::ASEEditor</class>
   <extends>QPlainTextEdit</extends>
   <header location="global">ASEEditor.hpp</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../ASE.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>actionSave_Script</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCompile</sender>
   <signal>triggered()</signal>
   <receiver>actionCompile_Script</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen_2</sender>
   <signal>triggered()</signal>
   <receiver>actionOpen</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDecompile</sender>
   <signal>triggered()</signal>
   <receiver>actionDecompile_Script</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLevelscript</sender>
   <signal>triggered()</signal>
   <receiver>actionDecompile_Levelscript</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
